Indicators/ParabolicStopAndReverseExtended.cs

```diff
@@ -0,0 +1,304 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using QuantConnect.Data.Market;
+
+namespace QuantConnect.Indicators
+{
+    /// <summary>
+    /// Parabolic SAR Extended Indicator 
+    /// Based on TA-Lib implementation
+    /// </summary>
+    public class ParabolicStopAndReverseExtended : BarIndicator, IIndicatorWarmUpPeriodProvider
+    {
+        private bool _isLong;
+        private IBaseDataBar _previousBar;
+        private decimal _sar;
+        private decimal _ep;
```

Comment:
`ep/sar/af/HasNegativeDM/etc` would be nice to expand these variable names to their full name, following the code style, makes it much easier to review/debug/understand 👍